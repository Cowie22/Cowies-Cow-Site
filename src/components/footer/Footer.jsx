import React, { memo } from 'react'
import { Link } from 'gatsby'
import { Container, Row, Col } from 'react-bootstrap'

import footerLogo from '../../assets/images/global/footer-logo.svg'
import arrowRight from '../../assets/images/global/right-arrow-yellow.svg'

const Footer = () => (
  <footer>
    <Container>
      <Row>
        <Col lg={{ span: 12, offset: 0 }}>
          <Row>
            <Col
              lg={{ span: 9, offset: 0, order: 1 }}
              xs={{ span: 12, offset: 0, order: 2 }}
            >
              <nav className='footer-nav-container'>
                <a
                  href='https://www.coherus.com/contact-us/'
                  target='_blank'
                  rel='noopener noreferrer'
                >
                  <img src={arrowRight} alt='' />
                  <h6 className='white bolder'>Contact Us</h6>
                </a>
                <a
                  href='https://www.coherus.com/privacy-policy/'
                  target='_blank'
                  rel='noopener noreferrer'
                >
                  <img src={arrowRight} alt='' />
                  <h6 className='white bolder'>Privacy Policy</h6>
                </a>
              </nav>
              <div className='footer-text-container'>
                <h6 className='white'>
                  Â© 2021 Coherus BioSciences, Inc. All rights reserved.
                  1021-IO-P031
                </h6>
              </div>
            </Col>
            <Col
              lg={{ span: 2, offset: 1, order: 2 }}
              xs={{ span: 6, offset: 3, order: 1 }}
            >
              <div className='footer-logo-container'>
                <a
                  href='https://www.coherus.com/'
                  target='_blank'
                  rel='noopener noreferrer'
                >
                  <img src={footerLogo} alt='Coherus BioSciences' />
                </a>
              </div>
            </Col>
          </Row>
        </Col>
      </Row>
    </Container>
  </footer>
)

export default memo(Footer)
