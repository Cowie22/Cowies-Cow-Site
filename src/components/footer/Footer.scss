footer {
  background: #F1F7F4;
  padding: 60px 0;
  nav {
    align-items: flex-end;
    height: fit-content;
    padding: 16px 0 32px 0;
    li {
      a {
        color: #A51890;
        transition: font-weight .4s;
        &:hover {
          cursor: pointer;
          color: #A51890;
          font-weight: bold;
          transition: font-weight .4s;
        }
      }
    }
  }
  .footer-text {
    padding-bottom: 16px;
  }
  .footer-logo-container {
    display: flex;
    justify-content: flex-end;
    align-items: flex-start;
  }
}