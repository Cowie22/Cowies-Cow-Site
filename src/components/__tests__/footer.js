import React from 'react'
import { render } from '@testing-library/react'
import Footer from '../footer/Footer'

const { debug, getByAltText, getByText, getByTestId, container } = render( <Footer /> )

const myovantLogo = getByAltText('Myovant Sciences logo')
const pfizerLogo = getByAltText('Pfizer logo')
const contactLink = getByText('Contact Us')
const privacyLink = getByText('Privacy Policy')
const termsLink = getByText('Terms of Use')
const siteMapLink = getByText('Site Map')

test('Footer links exist', () => {
  // debug()
  expect(myovantLogo).toBeTruthy()
  expect(pfizerLogo).toBeTruthy()
  expect(contactLink).toBeTruthy()
  expect(privacyLink).toBeTruthy()
  expect(termsLink).toBeTruthy()
  expect(siteMapLink).toBeTruthy()
})

test('Footer links have appropriate href', () => {
  expect(myovantLogo.parentElement.getAttribute('href')).toBe('https://www.myovant.com')
  expect(pfizerLogo.parentElement.getAttribute('href')).toBe('https://www.pfizer.com')
  expect(contactLink.getAttribute('href')).toBe('https://www.myovant.com/contact-us/')
  expect(privacyLink.getAttribute('href')).toBe('https://www.myovant.com/privacy-policy/')
  expect(termsLink.getAttribute('href')).toBe('https://www.myovant.com/terms-of-use/')
  expect(siteMapLink.getAttribute('href')).toBe('/sitemap/')
})

test('Footer snapshot matches', () => {
  // return
  // once the component is done, use snapshot
  expect(container.firstChild).toMatchSnapshot()
})