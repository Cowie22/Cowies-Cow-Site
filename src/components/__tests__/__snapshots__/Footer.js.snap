// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer snapshot matches 1`] = `null`;
